### PR TITLE
retry transient exceptions for PostgreSQL, fixes #1382

### LINKED ISSUE #1382: Need reasonable isTransientException implementation for Postgres
Currently PostgreSQLConnector inherits the default isTransientException implementation, which is just "return false". This means that potentially transient exceptions will never get retried.

### PR DESCRIPTION
**Pull request by xvrl**

### CODE REVIEW

File: extensions/postgresql-metadata-storage/src/main/java/io/druid/metadata/storage/postgresql/PostgreSQLConnector.java [hunk 1]
```diff
@@ -112,4 +114,16 @@ public Void withHandle(Handle handle) throws Exception
...
   public DBI getDBI() { return dbi; }
+
+  @Override
+  protected boolean isTransientException(Throwable e)
+  {
+    if(e instanceof SQLException) {
+      final String sqlState = ((SQLException) e).getSQLState();
+      // retry on connection errors and insufficient resources
+      // (see http://www.postgresql.org/docs/current/static/errcodes-appendix.html)
+      return sqlState.startsWith("08") || sqlState.startsWith("53");
```
Comment (cheddar) [inline extensions/postgresql-metadata-storage/src/main/java/io/druid/metadata/storage/postgresql/PostgreSQLConnector.java]: Of the "53" class of errors, I think only "53300" might be transient, no?

Comment (xvrl) [inline extensions/postgresql-metadata-storage/src/main/java/io/druid/metadata/storage/postgresql/PostgreSQLConnector.java]: out of memory or connection limit could also be transient I imagine

Comment (xvrl) [inline extensions/postgresql-metadata-storage/src/main/java/io/druid/metadata/storage/postgresql/PostgreSQLConnector.java]: I'm thinking more generally in terms of things that are not the client's fault and that might get resolved within a few retries, either manually or without intervention.

**Comment by drcrallen** [comment]
Can you add appropriate unit tests for this?

**Pull request merged**